spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/CommitMetadata.java

```diff
@@ -35,13 +40,18 @@ private CommitMetadata() {}
    * running the code wrapped as a caller, and any snapshot committed within the callable object
    * will be attached with the metadata defined in properties
    *
-   * @param properties extra commit metadata to attach to the snapshot committed within callable
+   * @param properties extra commit metadata to attach to the snapshot committed within callable.
+   *     The prefix will be removed for properties starting with {@link
+   *     SnapshotSummary#EXTRA_METADATA_PREFIX}
    * @param callable the code to be executed
    * @param exClass the expected type of exception which would be thrown from callable
    */
   public static <R, E extends Exception> R withCommitProperties(
       Map<String, String> properties, Callable<R> callable, Class<E> exClass) throws E {
-    COMMIT_PROPERTIES.set(properties);
+    Map<String, String> props = Maps.newHashMap();
+    properties.forEach((k, v) -> props.put(REMOVE_PREFIX.apply(k), v));
```

Comment:
It seems odd to use a Function instead of just calling `replace` inline here.

Reply:
fixed